helpers/Physics.h

```diff
@@ -0,0 +1,380 @@
+#ifndef _HELPERS_PHYSICS_H_
+#define _HELPERS_PHYSICS_H_
+
+#include "../include/z64.h"
+#include "Matrix.h"
+
+typedef struct {
+	s32   J; // Amount of strand limbs + 1
+	Vec3f headPos;
+	Vec3s headRot;
+	f32*  lengthList;
+	struct {
+		Vec3f scale;  // Gfx scale
+		u32   dlist;
+		u8    segID;  // For matrix
+		u8    noDraw; /* If there needs to be calculations
+		                 to get in position before drawing */
+	} gfx;
+	struct {
+		s32    numVec;      // num of how many vec are used from vecList
+		Vec3f* vecList;     // "collision" spheres
+		f32    dist;        // XYZ distance from limb to vecList member
+	} spheres;
+	struct {
+		Vec2f rotLimit;     // DEG, limits rot to next limb in main calc
+		f32   drawYawLimit; // DEG, limits rot on draw, smoothens output
+	} constraint;
+	
+	/* Physics */
+	f32 gravity;
+	f32 floor;        // world.pos.y, won't go through this
```

Comment:
```suggestion
	f32 floorY;       // world.pos.y, won't go through this
```
make it more obvious when reading it is a coordinate and not eg a floor collision polygon